java/code/src/com/redhat/rhn/domain/channel/test/ChannelFactoryTest.java

```diff
@@ -556,4 +557,27 @@ public void testFindAllByUserOrderByChild() throws Exception {
         assertEquals("a_child1", channels.get(2).getLabel());
         assertEquals("b_parent3", channels.get(3).getLabel());
     }
+
+    @Test
+    public void testCloneModulesMetadata() throws Exception {
+        User user = UserTestUtils.findNewUser("testUser", "testOrg" + this.getClass().getSimpleName());
+        Channel orig = ChannelTestUtils.createTestChannel(user);
+        Modules modules = new Modules("modules1.yaml", new Date());
```

Comment:
I think you can create with a date in the past, to make it more explicit the date should be kept.
I'm also not sure about default precision in date, or we can beeing use the same date in both create modules and hiding a possible error in case someone changes the logic.

Reply:
Agreed :+1: